peps/pep-9999.rst

```diff
@@ -0,0 +1,632 @@
+is a good pattern to have, or whether the API should expect the
+more specific type
+[`Issue 31 <https://github.com/capi-workgroup/problems/issues/31>`__].
+
+There are functions in the API that take concrete types, such as
+``PyDict_GetItemString`` which performs a dictionary lookup for a key
+specified as a c string rather than ``PyObject*``. At the same time,
+for ``PyDict_ContainsString`` it is not considered appropriate to
+add a concrete type alternative. The principle around this should
+be documented in the guidelines
+[`Issue 23 <https://github.com/capi-workgroup/problems/issues/23>`__].
+
+Implementation Flaws
+--------------------
+
+Below is a list of localized implementation flaws. Most of these can
+probably be fixed incrementally, if we choose to do so. They should,
+in any case, be avoided in any new API design.
+
+There are functions that don't follow the convention of
+returning ``0`` for success and ``-1`` for failure. For
+example, ``PyArg_ParseTuple`` returns 0 for success and
+non-zero for failure
+[`Issue 25 <https://github.com/capi-workgroup/problems/issues/25>`__].
+
+The macros ``Py_CLEAR`` and ``Py_SETREF`` access their arg more than
+once, so if the arg is an expression with side effects, they are
+duplicated
+[`Issue 3 <https://github.com/capi-workgroup/problems/issues/3>`__].
+
+The meaning of ``Py_SIZE`` depends on the type and is not always
+reliable
+[`Issue 10 <https://github.com/capi-workgroup/problems/issues/10>`__].
+
+**Naming**
+
+``PyLong`` and ``PyUnicode`` use names which don't match the python
+types they represent (int/str). This can be fixed in a new API
+[`Issue 14 <https://github.com/capi-workgroup/problems/issues/14>`__].
+
+There are identifiers in the API which are lacking a ``Py``/``_Py``
+prefix
+[`Issue 46 <https://github.com/capi-workgroup/problems/issues/46>`__].
+
+Some API function do not have the same behaviour as their Python
+equivalents.  The behaviour of ``PyIter_Next`` is different from
+``tp_iternext``.
+[`Issue 29 <https://github.com/capi-workgroup/problems/issues/29>`__].
+The behaviour of ``PySet_Contains`` is different from ``set.__contains__``
+[`Issue 6 <https://github.com/capi-workgroup/problems/issues/6>`__].
+
+The fact that ``PyArg_ParseTupleAndKeywords`` takes a non-const
+char* array as argument makes it more difficult to use.
+[`Issue 28 <https://github.com/capi-workgroup/problems/issues/28>`__].
+
+Python.h does not expose the whole API. Some headers (like marshal.h)
+are not included from Python.h.
+[`Issue 43 <https://github.com/capi-workgroup/problems/issues/43>`__].
+
+
+Missing Functionality
+---------------------
+
+This section consists of a list of feature requests, i.e., functionality
+that was identified as missing in the current C API.
+
+Debug Mode
+~~~~~~~~~~
+
+A debug mode that can be activated without recompilation and which
+activates various checks that can help detect various types of errors.
```

Comment:
```suggestion
activates various checks that can help detect various types of errors
```

Reply:
There is a full stop after the reference.